LanguageFeatures/Inline-classes/static_analysis_inline_class_A05_t02.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Assume that T1, .. Ts are types, and V resolves to an inline
+/// class declaration of the following form:
+///
+/// inline class V<X1 extends B1, .. Xs extends Bs> ... {
+///   final T id;
+///   V(this.id);
+///
+///   ... // Other members.
+/// }
+/// ...
+/// When s is zero, V<T1, .. Ts> simply stands for V, a non-generic inline type.
+/// When s is greater than zero, a raw occurrence V is treated like a raw type:
+/// Instantiation to bound is used to obtain the omitted type arguments
+///
+/// @description Checks that instantiation to bound is used to obtain the
+/// omitted type arguments
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/static_type_helper.dart";
+
+ class V<T extends V<T>> {
+  final T id;
+  V(this.id);
+}
+
+main() {
+  List<V> l = [];
+  l.expectStaticType<Exactly<List<V<V<dynamic>>>>>();
```

Comment:
This is tricky because the type `V<V<dynamic>>` should be an error in its own right (it is super-bounded). Same issue: the raw type `V` in the type `List<V>` should also be an error.

So I guess the only non-trivial test we can have for i2b is something like `<X extends List<Y>, Y extends num>` which should make it `<List<num>, num>`.

However, this test could be re-purposed to test that it is an error if the inline type isn't regular bounded.

Reply:
Hm... Ok, I'll change it.... But please note, there's also missed `inline` keyword for class `V` in this test but there's no any error. Is this an issue?

Reply:
I think this test has been taken out of the set of tests, or renamed --- I can't find anything like `V<V<dynamic>>` any more.

If the `inline` keyword is missing and we're working with a normal class then there is no problem in having super-bounded types, and hence no problem in having a type like `List<V<V<dynamic>>>`.

Reply:
Yes I removed it from here and going to add it later in another PR for `Subtyping` (and expect an error)